src/services/inputValidationService.js

```
@@ -0,0 +1,552 @@
+  sanitizeInput(input) {
+    if (!input || typeof input !== "string") {
+      return "";
+    }
+
+    return (
+      input
+        .trim()
+        // Remove null bytes
+        .replace(/\0/g, "")
+        // Remove control characters except newlines and tabs
+        .replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, "")
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)